app/init/network_manager.ts

```diff
@@ -0,0 +1,115 @@
+            ...LocalConfig.CustomRequestHeaders,
+        },
+        sessionConfiguration: {
+            allowsCellularAccess: true,
+            waitsForConnectivity: false,
+            timeoutIntervalForRequest: 30000,
+            timeoutIntervalForResource: 30000,
+            httpMaximumConnectionsPerHost: 10,
+            cancelRequestsOnUnauthorized: true,
+        },
+        retryPolicyConfiguration: {
+            type: 'exponential', // TODO: how do I used RetryTypes.EXPONENTIAL ?
+            retryLimit: 2,
+            exponentialBackoffBase: 2,
+            exponentialBackoffScale: 0.5,
+        },
+        requestAdapterConfiguration: {
+            bearerAuthTokenResponseHeader: 'token',
+        },
+    };
+
+    public init = async (serverCredentials: ServerCredential[]) => {
+        for await (const {serverUrl, token} of serverCredentials) {
+            try {
+                this.createClient(serverUrl, token);
+            } catch(error) {
+                console.log(error);
+            }
+        }
+    }
+
+    public invalidateClient = (serverUrl: string) => {
+        this.clients[serverUrl]?.client.invalidate();
+        delete this.clients[serverUrl];
+    }
+
+    public getClient = (serverUrl: string) => {
+        return this.clients[serverUrl] || this.createClient(serverUrl);
+    }
+
+    public createClient = async (serverUrl: string, token?: string) => {
+        const config = await this.buildConfig(token);
+        const {client} = await getOrCreateAPIClient(serverUrl, config, this.clientErrorEventHandler);
+        this.clients[serverUrl] = new Client(client, serverUrl);
+
+        return this.clients[serverUrl];
+    }
+
+    private buildConfig = async (token?: string) => {
+        const userAgent = await DeviceInfo.getUserAgent();
+        const headers: Record<string, any> = {
+            ...this.DEFAULT_CONFIG.headers,
+            [ClientConstants.HEADER_USER_AGENT]: userAgent,
+        };
+        if (token) {
+            headers[ClientConstants.HEADER_AUTH] = `${ClientConstants.HEADER_BEARER} ${token}`;
+        }
+
+        let config = {
+            ...this.DEFAULT_CONFIG,
+            headers,
+        }
+
+        if (ManagedApp.enabled) {
+            const managedConfig = await Emm.getManagedConfig();
+            if (managedConfig?.useVPN === 'true') {
+                config.sessionConfiguration.waitsForConnectivity = true;
+            }
+
+            if (managedConfig?.timeoutVPN) {
+                config.sessionConfiguration.timeoutIntervalForResource = parseInt(managedConfig.timeoutVPN, 10);
+            }
+        }
+
+        return config;
+    }
+
+    // TODO: typescript, APIClientErrorEventHandler and APIClientErrorEvent
+    private clientErrorEventHandler = (event) => {
+        // TODO: handle other errors
+        if (CLIENT_CERTIFICATE_IMPORT_ERROR_CODES.includes(event.errorCode)) {
+            // Emit CLIENT_CERTIFICATE_IMPORT_ERROR event
```

Comment:
as with the other event emitter comment, let's enable and use the RN event emitter